CA_DataUploaderLib/IOconf/IOconfSamplineRates.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace CA_DataUploaderLib.IOconf
+{
+    public class IOconfSamplingRates : IOconfState
+    {
+        public IOconfSamplingRates(string row, int lineNum) : base(row, lineNum, "SamplingRates")
+        {
+            format = "SamplingRates;MainLoop;VectorUploads";
+
+            var list = ToList();
+            if (list[0] != "SamplingRates" || list.Count != 3) throw new Exception("IOconfSamplingRates: wrong format: " + row);
+            
+            try
+            {
+                MainLoopDelay = (int)(1000 / double.Parse(list[1]));
+                var VectorUploadFreq = double.Parse(list[2]);
+                if(VectorUploadFreq > 2)
+                    throw new Exception("IOconfSamplingRates: too high upload frequency. Must be maximum 2 Hz: " + row);
+                VectorUploadDelay = (int)(1000 / VectorUploadFreq);
+            }
+            catch(Exception ex)
+            {
+                throw new Exception("IOconfMath: wrong format - expression: " + row, ex);
+            }
+        }
+
+        public int MainLoopDelay;
+        public int VectorUploadDelay;
```

Comment:
We will need to update the latest graphs sub sampling logic that assumes we upload approx. once x second. To do this, we could have the server extract it from the vector description.